jerry-core/jcontext/jcontext.h

```diff
@@ -187,6 +190,36 @@ extern jerry_hash_table_t jerry_global_hash_table;
 
 #endif /* !CONFIG_ECMA_LCACHE_DISABLE */
 
+#else /* JERRY_ENABLE_EXTERNAL_CONTEXT */
+
+#define JERRY_CONTEXT(field) (((jerry_context_t *) jerry_port_get_current_instance()->context_p)->field)
```

Comment:
> Yes, less indirection would help, that is why we have several macros rather than a single one (which would be enough theoretically, but in practice it is not efficient).

Right right. Sorry if I was unclear.
I'm basically saying we should have have separate `jerry_port_...` getters, one for each of these macros.